ludwig/utils/image_utils.py

```diff
@@ -107,14 +108,35 @@ def is_image_score(src_path, img_entry):
 
 
 @functools.lru_cache(maxsize=32)
-def read_image(img: Union[str, torch.Tensor], num_channels: Optional[int] = None) -> torch.Tensor:
-    """ Returns a tensor with CHW format.
+def read_image(img: Union[str, bytes, BytesIO, torch.Tensor], num_channels: Optional[int] = None) -> torch.Tensor:
+    """Returns a tensor with CHW format.
 
-    If num_channels is not provided, he image is read in unchanged format.
+    If num_channels is not provided, the image is read in unchanged format.
+    Returns None if the image could not be read.
     """
+    if isinstance(img, torch.Tensor):
+        return img
     if isinstance(img, str):
         return read_image_from_str(img, num_channels)
-    return img
+    if isinstance(img, bytes):
+        with BytesIO(img) as buffer:
+            buffer_view = buffer.getbuffer()
+            image_tensor = decode_image(
+                torch.frombuffer(buffer_view, dtype=torch.uint8))
+            del(buffer_view)
+            return image_tensor
+    if isinstance(img, BytesIO):
+        buffer_view = img.getbuffer()
+        try:
+            image_tensor = decode_image(
+                torch.frombuffer(buffer_view, dtype=torch.uint8))
+            del(buffer_view)
```

Comment:
is this basically memory cleanup?

Reply:
I think so. I'm not sure if it's completely necessary to manually delete the buffer here. @jimthompson5802 what was your reasoning for calling `del` manually?

Reply:
yes I was under the impression that garbage collector would kick in and do this for us, is that not the case?